Dockerfile

```diff
@@ -1,23 +1,22 @@
 FROM python:3.7-alpine
 LABEL maintainer="Jacob Tomlinson <jacob@tom.linson.uk>"
 
-RUN apk add --no-cache gcc musl-dev
-RUN mkdir -p /usr/src/app
+RUN apk update && apk add --no-cache gcc musl-dev git openssh-client
+
 WORKDIR /usr/src/app
 
 # Copy source
 COPY opsdroid opsdroid
-COPY setup.py setup.py
-COPY versioneer.py versioneer.py
 COPY setup.cfg setup.cfg
-COPY requirements.txt requirements.txt
 COPY README.md README.md
 COPY MANIFEST.in MANIFEST.in
+COPY requirements.txt requirements.txt
+COPY setup.py setup.py
+COPY versioneer.py versioneer.py
 
-RUN apk update && apk add --no-cache git openssh-client
-RUN pip3 install --upgrade pip
-RUN pip3 install --no-cache-dir --no-use-pep517 .
-RUN apk del gcc musl-dev
+RUN pip3 install --upgrade pip \
+&& pip3 install --no-cache-dir --no-use-pep517 . \
+&& apk del gcc musl-dev
```

Comment:
I would be tempted to move the `apk update` and `apk add` commands from line 4 to this block too. Some of these dependencies are only required to run the pip install, so it makes sense to remove them again in the same layer. This should really help with overall layer size.

Reply:
Good idea, I moved the apk operations down into this run command. Another layer down!